pkg/operator/ceph/object/shared_pools.go

```diff
@@ -151,9 +157,18 @@ func adjustZonePlacementPools(zone map[string]interface{}, spec cephv1.ObjectSha
 	return zone, nil
 }
 
+func getDefaultPlacementName(spec cephv1.ObjectSharedPoolsSpec) string {
+	for _, p := range spec.PoolPlacements {
+		if p.Default {
+			return p.Name
+		}
+	}
+	return defaultPlacementCephConfigName
+}
+
 func getDefaultMetadataPool(spec cephv1.ObjectSharedPoolsSpec) string {
 	for _, p := range spec.PoolPlacements {
-		if p.Name == defaultPlacementName {
+		if p.Default {
 			return p.MetadataPoolName
 		}
 	}
```

Comment:
Should we do a 2-part heuristic like this (bottom) here that allows a pool named `"default"` to assume default status in the absence of the default bool? I can understand how we might want to require the `default` bool to be the indicator, but I'm concerned that Ceph might treat `"default"` as the default if there is no bool set. 

1. If p.Default, return it
2. Else: if p.Name == defaultPlacementName, return it
3. Else: return spec.MetadataPoolName